src/wp-includes/option.php

```diff
@@ -823,7 +818,15 @@ function update_option( $option, $value, $autoload = null ) {
 	);
 
 	if ( null !== $autoload ) {
-		$update_args['autoload'] = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
+		$autoload                = sanitize_autoload( $autoload );
+		$update_args['autoload'] = $autoload;
+	} else {
+		$raw_autoload = $wpdb->get_var( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $option ) );
+		$allow_values = array( 'default-yes', 'default-no' );
+		if ( in_array( $raw_autoload, $allow_values, true ) ) {
+			$autoload                = sanitize_option_size( $option, $serialized_value );
+			$update_args['autoload'] = $autoload;
```

Comment:
See my previous feedback in https://github.com/WordPress/wordpress-develop/pull/4881#discussion_r1327785320, we should check here that the value is actually different from the current autoload value before setting it in `$update_args`.

Reply:
@felixarntz Do we need to do that? How does passing this to the update statement hurt anything?